unlock-app/src/services/web3Service.js

```diff
@@ -19,8 +19,9 @@ export const keyId = (lock, owner) => [lock, owner].join('-')
  * upstream objects.
  */
 export default class Web3Service extends EventEmitter {
-  constructor() {
+  constructor(myproviders = providers) {
     super()
+    this.providers = myproviders
```

Comment:
Maybe rename is `this.availableProviders` as it gives a bit more context about what that variable holds?